pkg/providers/pricing/suite_test.go

```diff
@@ -197,6 +204,7 @@ var _ = Describe("Pricing", func() {
 		price, ok := awsEnv.PricingProvider.SpotPrice("c98.large", "test-zone-1a")
 		Expect(ok).To(BeTrue())
 		Expect(price).To(BeNumerically("==", 1.20))
+		Expect(getPricingEstimateMetricValue("c98.large", ec2.UsageClassTypeSpot, "test-zone-1a")).To(BeNumerically("==", 1.20))
```

Comment:
We have a metric getter function in `karpenter-core` that you can leverage here called `FindMetricWithLabelValues` instead of having to write your own